src/rpc_service.c

```diff
@@ -160,12 +173,20 @@ rpc_context_create(void)
 void
 rpc_context_free(rpc_context_t context)
 {
+	struct emit_item *item;
 
 	if (context == NULL)
 		return;
 
 	g_thread_pool_free(context->rcx_threadpool, true, true);
 	rpc_instance_free(context->rcx_root);
+
+	item = g_malloc(sizeof (*item));
+        item->context = NULL;
+        g_async_queue_push(context->rcx_emit_queue, item);
```

Comment:
Some indentation issues here

Reply:
Ah, thats what I get for mixing languages. I'll fix it!